assessment_factory/assessment_factory.py

```diff
@@ -1,65 +1,238 @@
 """TO-DO: Write a description of what this XBlock is."""
 
+import json
 import pkg_resources
+import urllib
+import logging
+import hashlib
+import mimetypes
+import os
+
+from functools import partial
+
+from django.core.files import File
+from django.core.files.storage import default_storage
+from PIL import Image
+from webob.response import Response
 
 from xblock.core import XBlock
-from xblock.fields import Scope, Integer, String
+from xblock.fields import Scope, Integer, String, Float, Dict, List, Boolean
 from xblock.fragment import Fragment
+from xblockutils.resources import ResourceLoader
+from xblockutils.settings import XBlockWithSettingsMixin, ThemableXBlockMixin
 
+loader = ResourceLoader(__name__)
+BLOCK_SIZE = 8 * 1024
 
-class AssessmentFactoryXBlock(XBlock):
+class AssessmentFactoryXBlock(XBlock, XBlockWithSettingsMixin, ThemableXBlockMixin):
     """
     TO-DO: document what your XBlock does.
     """
 
     # Fields are defined on the class.  You can access them in your code as
     # self.<fieldname>.
 
-    # TO-DO: delete count, and define your own fields.
     display_name = String(
         display_name="Title",
         default="Assessment Factory",        
         scope=Scope.settings,
         help="The title of the Assessment Factory problem. The title is displayed to learners.",
     )
 
-    count = Integer(
-        default=0, scope=Scope.user_state,
-        help="A simple counter, to show something happening",
+    weight = Float(
+        display_name="Weight",
+        help="The maximum score the learner can receive for the problem",
+        scope=Scope.settings,
+        default=100,
+    )
+
+    studio_assignment = Dict(
+        display_name="Studio Assignment Dictionary",
+        default={},
+        scope=Scope.settings,
+        help="Dictionary containing items and categories that are set in studio."
+    )
+
+    item_state = Dict(
+        display_name="Item State",
+        default={},
+        scope=Scope.user_state,
+        help="Dictionary containing item state information."
+    )
+
+    is_graded = Boolean(
+        help="Indicates whether the problem is graded.",
+        scope=Scope.user_state,
+        default=False,
+    )
+
+    retry_count = Integer(
+        display_name="Retry Count",
+        default=3,
+        scope=Scope.user_state,
+        help="Number of retries user is allowed to use."
+    )
+
+    incorrect_items = List(
+        display_name="Incorrect Items List",
+        default=[],
+        scope=Scope.user_state,
+        help="List of incorrect items used for retrying."
     )
 
+    correct_items = List(
+        display_name="Correct Items List",
+        default=[],
+        scope=Scope.user_state,
+        help="List of correct items used for retrying."
+    )
+
+    # Needed for grading
+    has_score = True
```

Comment:
Ovo sam u onom drag and drop xblocku vidia, mislia san da samo treba stavit ovako, ako se zeli ocjenjivanje